mmv1/third_party/terraform/services/resourcemanager/ephemeral_google_client_config.go

```diff
@@ -0,0 +1,142 @@
+			},
+			"project": schema.StringAttribute{
+				Description:         "The ID of the project to apply any resources to.",
+				MarkdownDescription: "The ID of the project to apply any resources to.",
+				Computed:            true,
+			},
+			"region": schema.StringAttribute{
+				Description:         "The region to operate under.",
+				MarkdownDescription: "The region to operate under.",
+				Computed:            true,
+			},
+			"zone": schema.StringAttribute{
+				Description:         "The zone to operate under.",
+				MarkdownDescription: "The zone to operate under.",
+				Computed:            true,
+			},
+			"access_token": schema.StringAttribute{
+				Description:         "The OAuth2 access token used by the client to authenticate against the Google Cloud API.",
+				MarkdownDescription: "The OAuth2 access token used by the client to authenticate against the Google Cloud API.",
+				Computed:            true,
+				Sensitive:           true,
+			},
+			"default_labels": schema.MapAttribute{
+				Description:         "The default labels configured on the provider.",
+				MarkdownDescription: "The default labels configured on the provider.",
+				Computed:            true,
+				ElementType:         types.StringType,
+			},
+		},
+	}
+}
+
+func (e *GoogleClientConfigEphemeralResource) Configure(ctx context.Context, req ephemeral.ConfigureRequest, resp *ephemeral.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	p, ok := req.ProviderData.(*transport_tpg.Config)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Ephemeral Resource Configure Type",
+			fmt.Sprintf("Expected *transport_tpg.Config, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	// Required for accessing project, region, zone and tokenSource
+	e.providerConfig = p
+}
+
+func (e *GoogleClientConfigEphemeralResource) Open(ctx context.Context, req ephemeral.OpenRequest, resp *ephemeral.OpenResponse) {
+	var data GoogleClientConfigEphemeralModel
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	data.Id = types.StringValue(fmt.Sprintf("projects/%s/regions/%s/zones/%s", e.providerConfig.Project, e.providerConfig.Region, e.providerConfig.Zone))
+	data.Project = types.StringValue(e.providerConfig.Project)
+	data.Region = types.StringValue(e.providerConfig.Region)
+	data.Zone = types.StringValue(e.providerConfig.Zone)
+
+	// Convert default labels from SDK type system to plugin-framework data type
+	m := map[string]*string{}
+	for k, v := range e.providerConfig.DefaultLabels {
+		// m[k] = types.StringValue(v)
+		val := v
+		m[k] = &val
+	}
+	dls, diags := types.MapValueFrom(ctx, types.StringType, m)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	data.DefaultLabels = dls
```

Comment:
im confused about this new map being created, it uses the `DefaultLabels` which is already a map and loops through in order to store the key and value inside the new map `m` yet there isn't a change that's occurred? I imagine this is from a first iteration of this PR since i believe you can just utilize `e.providerConfig.DefaultLabels` as the argument in `types.MapValueFrom()`

Reply:
I copied this from the `google_client_config` datasource's implementation without giving it much thought. It seems like we can indeed pass `e.providerConfig.DefaultLabels` to `types.MapValueFrom()` directly, as you say. I ran a few local sanity checks and it seems to work the same.

I've updated the file I added as well as a couple other places in the codebase where we did the same thing.

Reply:
makes sense to do, i noticed that `google_client_config` data source written using the `plugin-framework`.

feel free to open a PR that also updates the logic in the data-source so that it aligns here.